src/module/migration/migrations/884-unified-spellcasting.ts

```diff
@@ -0,0 +1,30 @@
+import { ActorSourcePF2e } from "@actor/data/index.ts";
+import { MigrationBase } from "../base.ts";
+import { CharacterSystemSource } from "@actor/character/data.ts";
+import { recursiveReplaceString } from "@util";
+import { ItemSourcePF2e } from "@item/base/data/index.ts";
+
+export class Migration884UnifiedSpellcasting extends MigrationBase {
+    static override version = 0.884;
+
+    override async updateActor(source: ActorSourcePF2e): Promise<void> {
+        if (source.type !== "character") return;
+
+        // traditions was prepared data, but remove it just in case someone was naughty
+        const profs: CharacterProficienciesOld | null = source.system.proficiencies ?? null;
+        if (profs?.traditions) {
+            profs["-=traditions"] = null;
+        }
+    }
+
+    override async updateItem(source: ItemSourcePF2e): Promise<void> {
+        source.system.rules = recursiveReplaceString(source.system.rules, (value) =>
+            value.replace(/^system\.proficiencies\.traditions\.(.*)\.(\w+)$/, "system.proficiencies.spellcasting.$2"),
```

Comment:
```suggestion
            value.replace(/^system\.proficiencies\.traditions\.[^.]+\.(\w+)$/, "system.proficiencies.spellcasting.$1"),
```

Reply:
This will fail to catch the {actor|blah.blah} stuff. We have to be greedy unfortunately.

Reply:
Is "blah.blah" going to be a valid property path following "system.proficiencies.spellcasting"?